landlab/components/flow_accum/lossy_flow_accumulator.py

```diff
@@ -0,0 +1,464 @@
+        "drainage_area": "m**2",
+        "surface_water__discharge": "m**3/s",
+        "surface_water__discharge_loss": "m**3/s",
+        "flow__upstream_node_order": "-",
+        "flow__data_structure_delta": "-",
+        "flow__data_structure_D": "-",
+        "flow__nodes_not_in_stack": "-",
+    }
+
+    _var_mapping = {
+        "topographic__elevation": "node",
+        "flow__receiver_node": "node",
+        "water__unit_flux_in": "node",
+        "drainage_area": "node",
+        "surface_water__discharge": "node",
+        "surface_water__discharge_loss": "node",
+        "flow__upstream_node_order": "node",
+        "flow__nodes_not_in_stack": "grid",
+        "flow__data_structure_delta": "node",
+        "flow__data_structure_D": "link",
+    }
+    _var_doc = {
+        "topographic__elevation": "Land surface topographic elevation",
+        "flow__receiver_node": "Node array of receivers (node that "
+        + "receives flow from current node)",
+        "drainage_area": "Upstream accumulated surface area contributing "
+        + "to the node's discharge",
+        "surface_water__discharge": "Discharge of water through each node",
+        "surface_water__discharge_loss": "Total volume of water per second "
+        + "lost during all flow out of the node",
+        "water__unit_flux_in": "External volume water per area per time "
+        + "input to each node (e.g., rainfall rate)",
+        "flow__upstream_node_order": "Node array containing "
+        + "downstream-to-upstream ordered list of node IDs",
+        "flow__data_structure_delta": "Node array containing the "
+        + "elements delta[1:] of the data structure 'delta' used for "
+        + "construction of the downstream-to-upstream node array",
+        "flow__data_structure_D": "Link array containing the data structure "
+        + "D used for construction of the downstream-to-upstream node array",
+        "flow__nodes_not_in_stack": "Boolean value indicating if there "
+        + "are any nodes that have not yet been added to the stack stored "
+        + "in flow__upstream_node_order.",
+    }
+
+    def __init__(
+        self,
+        grid,
+        surface="topographic__elevation",
+        flow_director="FlowDirectorSteepest",
+        runoff_rate=None,
+        depression_finder=None,
+        loss_function=None,
+        **kwargs
+    ):
+        """
+        Initialize the FlowAccumulator component.
+
+        Saves the grid, tests grid type, tests imput types and compatability
+        for the flow_director and depression_finder keyword arguments, tests
+        the argument of runoff_rate, and initializes new fields.
+        """
+        super(LossyFlowAccumulator, self).__init__(
+            grid,
+            surface=surface,
+            flow_director=flow_director,
+            runoff_rate=runoff_rate,
+            depression_finder=depression_finder,
+            **kwargs
+        )
+
+        if loss_function is not None:
```

Comment:
@SiccarPoint and @mcflugen 

Once this PR is merged, both LossyFlowAccumulator _and_ Lithology will do function argument and return checking. Does it make sense to move the function checking capabilities for both of them to `landlab.utilities`?